src/connector.rs

```diff
@@ -4,12 +4,12 @@ use std::sync::Arc;
 use std::task::{Context, Poll};
 use std::{fmt, io};
 
-use hyper::client::connect::Connection;
-use hyper::service::Service;
-use hyper::Uri;
+use http::Uri;
+use hyper::rt::{Read, Write};
+use hyper_util::rt::TokioIo;
 use pki_types::ServerName;
-use tokio::io::{AsyncRead, AsyncWrite};
 use tokio_rustls::TlsConnector;
+use tower_service::Service;
```

Comment:
The Hyper 1.0 [upgrade guide](https://hyper.rs/guides/1/upgrading/) says:
> Previously, hyper depended on tower for the Service trait. Because tower is not yet 1.0, hyper could not publicly depend on it. So, it’s Service trait and the helper function service_fn are defined in hyper::service.

I'm not clear on why we need to use a direct dependency on `tower_service` and its `Service` trait here instead of the [hyper::service::Service](https://docs.rs/hyper/latest/hyper/service/trait.Service.html) trait. Can you elaborate on this part of the changeset?

Reply:
Because hyper-util has different trait bounds. See https://docs.rs/hyper-util/latest/hyper_util/client/legacy/connect/trait.Connect.html, it requires the `tower_service` `Service<Uri>` trait to be implemented for the connector and not hyper's.